### DISCUSSION TITLE
use files from your extension in task providers

### DISCUSSION THREAD

**Discussion by Judimax** [microsoft/vscode-discussions#648] [Extension Development QnA] [answered]
Greetings I am currently developing a vscode extension and I am trying to use a file in my vscode extension as part of a task. I first need the extension to be able to see the file in its own directory (in the out folder where the extension code is). Then use it in a vscode.ShellExecution to run a task does anyone know how to go about this google search and AI results are returning results as if  I am not developing a vscode extension

**Comment by r3m0t** [comment]
`ExtensionContext.extensionUri` and `Uri.joinPath`

**Comment by Judimax** [comment]
Where is the documentation for this?

**Reply by gjsjohnmurray** [reply, marked as answer]
Also at https://code.visualstudio.com/api/references/vscode-api#ExtensionContext

Your `activate` function receives an instance of `vscode.ExtensionContext` as the `context` parameter, telling you a lot about yourself.

**Reply by Judimax** [reply]
woah woah  woah there, I get it you work at microsoft and I love your product no need to gloat, besides this is not mentioned in the tutorials even you did not provide a tutorial link. This is an example of being clever and not clear. Context is not a regular word in your average programmers dictionary, and vscode extension are an auxillary development tool, not an application that makes money, so while things are in the familar tense on the vscode internal team, its very abstract on the outside. Just my two cents on the great documentation you have

**Reply by gjsjohnmurray** [reply]
I don't work for Microsoft, and I contribute here in the spirit of open source.

**Comment by r3m0t** [comment]
https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts

There's an example here
https://code.visualstudio.com/api/extension-guides/webview
